src/heap_type.js

```diff
@@ -0,0 +1,137 @@
+//Provides: caml_binaryen_heap_type_ext
+//Requires: Binaryen
+function caml_binaryen_heap_type_ext() {
+  return Binaryen._BinaryenHeapTypeExt();
+}
+
+//Provides: caml_binaryen_heap_type_func
+//Requires: Binaryen
+function caml_binaryen_heap_type_func() {
+  return Binaryen._BinaryenHeapTypeFunc();
+}
+
+//Provides: caml_binaryen_heap_type_any
+//Requires: Binaryen
+function caml_binaryen_heap_type_any() {
+  return Binaryen._BinaryenHeapTypeAny();
+}
+
+//Provides: caml_binaryen_heap_type_eq
+//Requires: Binaryen
+function caml_binaryen_heap_type_eq() {
+  return Binaryen._BinaryenHeapTypeEq();
+}
+
+//Provides: caml_binaryen_heap_type_i31
+//Requires: Binaryen
+function caml_binaryen_heap_type_i31() {
+  return Binaryen._BinaryenHeapTypeI31();
+}
+
+//Provides: caml_binaryen_heap_type_data
+//Requires: Binaryen
+function caml_binaryen_heap_type_data() {
+  return Binaryen._BinaryenHeapTypeData();
+}
+
+//Provides: caml_binaryen_heap_type_array
+//Requires: Binaryen
+function caml_binaryen_heap_type_array() {
+  return Binaryen._BinaryenHeapTypeArray();
+}
+
+//Provides: caml_binaryen_heap_type_string
+//Requires: Binaryen
+function caml_binaryen_heap_type_string() {
+  return Binaryen._BinaryenHeapTypeString();
+}
+
+//Provides: caml_binaryen_heap_type_stringview_wtf8
+//Requires: Binaryen
+function caml_binaryen_heap_type_stringview_wtf8() {
+  return Binaryen._BinaryenHeapTypeStringviewWTF8();
+}
+
+//Provides: caml_binaryen_heap_type_stringview_wtf16
+//Requires: Binaryen
+function caml_binaryen_heap_type_stringview_wtf16() {
+  return Binaryen._BinaryenHeapTypeStringviewWTF16();
+}
+
+//Provides: caml_binaryen_heap_type_stringview_iter
+//Requires: Binaryen
+function caml_binaryen_heap_type_stringview_iter() {
+  return Binaryen._BinaryenHeapTypeStringviewIter();
+}
+
+//Provides: caml_binaryen_heap_type_none
+//Requires: Binaryen
+function caml_binaryen_heap_type_none() {
+  return Binaryen._BinaryenHeapTypeNone();
+}
+
+//Provides: caml_binaryen_heap_type_noext
+//Requires: Binaryen
+function caml_binaryen_heap_type_noext() {
+  return Binaryen._BinaryenHeapTypeNoext();
+}
+
+//Provides: caml_binaryen_heap_type_nofunc
+//Requires: Binaryen
+function caml_binaryen_heap_type_nofunc() {
+  return Binaryen._BinaryenHeapTypeNofunc();
+}
+
+//Provides: caml_binaryen_heap_type_is_basic
+//Requires: Binaryen
+function caml_binaryen_heap_type_is_basic(ty) {
+  // TODO: Verify that binaryen returns an int which jsoo will understand as bool
```

Comment:
Have these been verified?

Reply:
I tested this one specifically. Other ones don't seem to work by just passing a newly created HeapType; looking at the JS function contents, it looks like there is a lookup of the type in some other data structure.

Let's call these good.